scripts/reboot

```diff
@@ -10,7 +10,7 @@ if [ -e $VMCORE_FILE -a -s $VMCORE_FILE ]; then
         echo "We have a /proc/vmcore, then we just kdump'ed"
         echo "User issued 'kdump' command [User: kdump, Time: ${REBOOT_TIME}]" > ${REBOOT_CAUSE_FILE}
         sync
-        PLATFORM=$(grep -oP 'platform=\K\S+' /proc/cmdline)
+        PLATFORM=$(grep -oP 'sonic_platform=\K\S+' /proc/cmdline)
```

Comment:
I have curious why we cannot use sonic-cfggen to get the platform as line 24?

Reply:
When we are running in the crash kernel context, we do not want to access databases or libraries or anything more involving. The idea is to do as little work as possible to dump the crash kernel and reboot. For that reason, we are passing the platform name as part of the command line.

Reply:
@lguohan can you please approve and merge these changes. We need to move the submodule pointer in sonic-buildimage for all the changes to work.